### PR TITLE
Makes the GraphQL user type reusable.

### LINKED ISSUE #15540: The CustomUserSettings in the graphql me field is registered incorrectly
The type registration error occurred because the AddField method in the UserType type had a method error when registering the type

The fields in the red box below are not of the type for `CustomUserSettings`, they are actually `CustomSettings`.

image

this PR https://github.com/OrchardCMS/OrchardCore/pull/15389, will fix the problem

### PR DESCRIPTION
**Pull request by gvkries**
Enhances how the GraphQL `UserType` is build to make it more reliable when reused e.g. by https://github.com/OrchardCMS/OrchardCore/pull/15389 The `UserType` does not depend on the schema builder to be populated any longer.

Also fixes #15540

### CODE REVIEW

**Comment by hishamco** [comment]
@hyzx86 your review please

**Comment by hyzx86** [comment]
Hi @hishamco , When you review PR daily, do you manually check out contributors' repositories or do you have any tools to assist in this process?

**Comment by hishamco** [comment]
You can review by looking to the code, but sometimes you might need to try the change that made

**Comment by hyzx86** [comment]
> You can review by looking to the code, but sometimes you might need to try the change that made

Yes, I need to try this PR

**Comment by hyzx86** [comment]
I feel fine with the code and functionality.

but there is one small issue that has nothing to do with this PR
I haven't enabled `UserCustomSettings` yet, but it's still available in the me api
image

**Comment by gvkries** [comment]
> I feel fine with the code and functionality.
> 
> but there is one small issue : I haven't enabled `UserCustomSettings` yet, but it's still available in the `me` field image

Yes, it looks at the stereotype only. Whether the feature is actually enabled is not checked (and was not before this PR). Could be improved, but I don't think this is important.

**Comment by hishamco** [comment]
@MikeAlhayek anything to add here?

**Comment by Piedone** [comment]
@hishamco please merge if you approve.

**Comment by Piedone** [comment]
@hyzx86 for conducting reviews, see [these yet unmerged tips](https://github.com/OrchardCMS/OrchardCore/pull/15706/files#diff-99b2eb2d1d7935ad6f2719d19ac3434df09a71cc15667ea77772097aa37c80b8) I added.

File: src/OrchardCore.Modules/OrchardCore.Users/GraphQL/CurrentUserQuery.cs [hunk 1]
```diff
@@ -21,45 +17,23 @@ namespace OrchardCore.Users.GraphQL;
...
-
-            userType.AddField(schema, typeDefinition);
-        }
-
         var currentUserField = new FieldType
         {
             Name = "me",
             Description = S["Gets the currently authenticated user."],
-            ResolvedType = userType,
+            Type = typeof(UserType),
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Users/GraphQL/CurrentUserQuery.cs:36]: Where the `UserType` is defined?

File: src/OrchardCore.Modules/OrchardCore.Users/GraphQL/UserType.cs [hunk 2]
```diff
@@ -18,39 +16,38 @@ public UserType(IStringLocalizer<UserType> localizer)
...
-                    var customUserSettingsService = context.RequestServices!.GetRequiredService<CustomUserSettingsService>();
-                    var settingsType = await customUserSettingsService.GetSettingsTypeAsync(context.FieldDefinition.ResolvedType.Name);
-
-                    return await customUserSettingsService.GetSettingsAsync(user, settingsType);
-                }
+                    // We don't want to create an empty content item if it does not exist.
+                    if (context.Source is User user &&
+                        user.Properties.ContainsKey(context.FieldDefinition.ResolvedType.Name))
+                    {
+                        var customUserSettingsService = context.RequestServices!.GetRequiredService<CustomUserSettingsService>();
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Users/GraphQL/UserType.cs:41]: Why `RequestServices!`

**Comment by hishamco** [comment]
@hyzx86 your review please

File: src/OrchardCore.Modules/OrchardCore.Users/GraphQL/CurrentUserQuery.cs (hunk 1 above)
Comment (gvkries) [reply src/OrchardCore.Modules/OrchardCore.Users/GraphQL/CurrentUserQuery.cs:36]: https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Users/GraphQL/UserType.cs

File: src/OrchardCore.Modules/OrchardCore.Users/GraphQL/UserType.cs (hunk 2 above)
Comment (gvkries) [reply src/OrchardCore.Modules/OrchardCore.Users/GraphQL/UserType.cs:41]: It's defined as nullable, but can't be in our case as far as I know. If you don't want nullable annotations being used, I can of cause remove them.

**Review by hyzx86 — APPROVED** [review]
Looks good to me

**Comment by hyzx86** [comment]
@hishamco ,  Merge it please 😎 , the PR https://github.com/OrchardCMS/OrchardCore/pull/15389 need  this

**Comment by hishamco** [comment]
Just waiting for a last note then I will merge

File: src/OrchardCore.Modules/OrchardCore.Users/GraphQL/UserType.cs (hunk 2 above)
Comment (gvkries) [reply src/OrchardCore.Modules/OrchardCore.Users/GraphQL/UserType.cs]: Okay, I removed it. But just for completeness, it was already used before this PR 🤷‍♂️

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Users/GraphQL/UserType.cs]: I see, just leave my comment

**Review by hishamco — APPROVED** [review]

**Pull request merged**